### DISCUSSION TITLE
Webservice put related module record

### DISCUSSION THREAD

**Discussion by noelhunter** [YetiForceCompany/YetiForceCRM#15239] [How to] [answered]
I am able to GET related module records in the webservice restapi no problem, but I cannot see in postman how to create a relationship when I POST or PUT an opportunity.  Specifically, when I add an opportunity, I want to set the related account.  Can someone point to the documentation or an example?

**Comment by rskrzypczak** [comment, marked as answer]
The opportunity is related to the account by the 1-M relationship, so all you need to do is complete the reference field.
![image](https://user-images.githubusercontent.com/10221999/135493480-7470aed3-41d3-4db7-a918-42abe158e698.png)

**Reply by noelhunter** [reply]
Thanks-- that helped.  I am successfully creating opportunities, but the relationship is being skipped.  The opportunity is created otherwise.

`----------- _POST -----------
Array
(
    [subject] => test
    [related_to] => 159
    [description] => test
)

----------- Request payload -----------

-------------  Response  -----  2021-09-30 12:46:18  ------
Status: 200
REQUEST_METHOD: POST
REQUEST_URI: /webservice.php?_container=RestApi&module=SSalesProcesses&action=Record
QUERY_STRING: _container=RestApi&module=SSalesProcesses&action=Record
PATH_INFO:
----------- Response data -----------
Array
(
    [status] => 1
    [result] => Array
        (
            [id] => 2877
            [name] => test
            [skippedData] => Array
                (
                    [0] => related_to
                )

        )

)`

**Reply by noelhunter** [reply]
I figured it out.  If I add a custom field with the ID ssalesprocesses_contactid to the opportunities, then I can send related_to records successfully.  Thanks for your help.

**Comment by noelhunter** [comment]
Now I am trying to send a related product with an opportunity or lead.  I can't seem to figure out the format for the JSON to send that relationship (a many to many?)

**Reply by rskrzypczak** [reply]
Take a look here: https://yetiforce.com/en/knowledge-base/documentation/developer-documentation/item/how-to-add-a-document-with-an-attachment-via-rest-api
so you should just add three parameters to the query:
```
"relationOperation": true,
"sourceModule: SOURCE_MODULE,
"sourceRecord": SOURCE_RECORD_ID
```

**Reply by noelhunter** [reply]
Thank you!

**Reply by noelhunter** [reply]
When I put, the relationship IS added, however, the result string indicates that the data was skipped.  I am including an example of what I am putting.

`{{baseUrl}}/webservice/RestApi/Leads/Record/5202
{
    "relationOperation": "true",
    "sourceModule" :"Products",
	"sourceRecord" : 4221
}

Response:

{
    "status": 1,
    "result": {
        "id": 5202,
        "name": "redacted",
        "skippedData": {
            "0": "relationOperation",
            "1": "sourceModule",
            "2": "sourceRecord"
        }
    }
}`

**Reply by rskrzypczak** [reply]
You can ignore it.
skippedData applies to the module fields, and since there is no field in the module Leads for any of these variables, they are automatically placed in skippedData.
